### PR TITLE
[Issue 185] Allow adding labels to issues via a file in the $SHARED_DIR

### LINKED ISSUE #185: [Enhancement] Allow adding labels to issues via a file in the $SHARED_DIR
OpenShift CI has a shared directory that persists between the steps run in a pipeline. As part of the product to product pipeline, we would like to dynamically add labels based on the OpenShift version and product versions installed. Our proposed solution to this would be to add functionality to firewatch to read a list of labels found in a specific file in the $SHARED_DIR and add those labels to any new ticket created.

This functionality can be used for more than the use case outlined above, but essentially the flow would be:

1. Write any additional labels to `$SHARED_DIR/firewatch-additional-labels`, this can be done over multiple steps throughout the job. They should be separated by a new line (or some other delimiter). 
3. When firewatch is executed, prior to creating any bugs, read the list of labels and ensure they are all applied to any new issue created (along with all the pre-configured labels in the firewatch config).

### PR DESCRIPTION
**Pull request by calebevans**
- Adds functionality to include any labels listen in the `$SHARED_DIR/firewatch-additional-labels` file in OpenShift CI.
    - Labels should be separated by a new line.
- Add tests for new functionality.
- Updates test job, old one seems to be out of date and causing tests to fail.

Example:
1. Add a label to the file in OpenShift CI: `echo "some-label" >>  $SHARED_DIR/firewatch-additional-labels`.
2. Observe that any new ticket created will include the `some-label` label.

Closes #185

### CODE REVIEW

**Comment by redhat-qe-bot1** [comment]
Report bugs in [Issues](https://github.com/myakove/github-webhook-server/issues)

The following are automatically added:
 * Add reviewers from OWNER file (in the root of the repository) under reviewers section.
 * Set PR size label.
 * New issue is created for the PR. (Closed when PR is merged/closed)
 * Run [pre-commit](https://pre-commit.ci/) if `.pre-commit-config.yaml` exists in the repo.

Available user actions:
 * To mark PR as WIP comment `/wip` to the PR, To remove it from the PR comment `/wip cancel` to the PR.
 * To block merging of PR comment `/hold`, To un-block merging of PR comment `/hold cancel`.
 * To mark PR as verified comment `/verified` to the PR, to un-verify comment `/verified cancel` to the PR.
        verified label removed on each new commit push.
 * To cherry pick a merged PR comment `/cherry-pick <target branch to cherry-pick to>` in the PR.
    * Multiple target branches can be cherry-picked, separated by spaces. (`/cherry-pick branch1 branch2`)
    * Cherry-pick will be started when PR is merged
 * To build and push container image command `/build-and-push-container` in the PR (tag will be the PR number).
 * To add a label by comment use `/<label name>`, to remove, use `/<label name> cancel`
 * To assign reviewers based on OWNERS file use `/assign-reviewers`
 * To check if PR can be merged use `/check-can-merge`

<details>
<summary>Supported /retest check runs</summary>

 * This repository does not support retest actions
</details>

<details>
<summary>Supported labels</summary>

 * hold
 * verified
 * wip
 * lgtm

</details>

**Comment by dfrazzette** [comment]
@calebevans - Is it worth updating the Approvers in the repo's [OWNERS](https://github.com/RedHatQE/firewatch/blob/main/OWNERS) list? The `can-be-merged`
check is failing with: 
`Missing lgtm/approved from approvers ['calebevans', 'btjd', 'ascerra', 'myakove', 'rnetser']`

File: src/report/report.py
Comment (myakove) [inline src/report/report.py:574]: since it's  path please use `os.path.join`

File: tests/unittests/functions/report/test_firewatch_functions_report_get_issue_labels.py
Comment (myakove) [inline tests/unittests/functions/report/test_firewatch_functions_report_get_issue_labels.py:43]: I may be missing here something but why assert twice `test-1`

**Review by myakove — CHANGES_REQUESTED** [review]

File: src/report/report.py
Comment (oharan2) [inline src/report/report.py:574]: I have one more approach to suggest, I'm not sure if it's better but worth mention:

Shouldn't we set the filepath as a cli input with click and then handle setting it's value with the `$SHARED_DIR/firewatch-additional-labels` file within the openshift-ci end?

I.e. Openshift-Ci will have a ref that handles extracting all the labels in bash into a file, then set the `jira_additional_labels_filepath` value in the `firewatch-report-issues` script

Comment (oharan2) [reply src/report/report.py:574]: Another thought, can't we handle it completly on Openshift-Ci end by modifying the lables env var and extending it in a dedicated ref?

I mean user will set the config file with:
```
FIREWATCH_DEFAULT_JIRA_ADDITIONAL_LABELS: '["4.15-lp","rosa-classic-lp"]'
```
And then one more ref will parse and update the env var in bash to be set with:
```      
FIREWATCH_DEFAULT_JIRA_ADDITIONAL_LABELS: '["4.15-lp","rosa-classic-lp", "added-label","one-more-label"...]'
```

File: tests/unittests/functions/report/test_firewatch_functions_report_get_issue_labels.py
Comment (calebevans) [reply tests/unittests/functions/report/test_firewatch_functions_report_get_issue_labels.py:43]: You're right, that was a typo. Thanks for catching :)

File: src/report/report.py
Comment (calebevans) [reply src/report/report.py:574]: I like the idea of setting it in the CLI and passing the file in OpenShift CI. I think we should avoide adding another step to parse the file, though I think it would work. I will make changes to pass the file in the CLI :)

Comment (calebevans) [reply src/report/report.py]: Ended up taking the path from a cli argument

File: src/commands/report.py
Comment (myakove) [inline src/commands/report.py:123]: `required=False` this is the default, no need to set it.

**Review by myakove — CHANGES_REQUESTED** [review]

File: src/report/report.py
Comment (rnetser) [inline src/report/report.py:603]: why is `if line.strip()` needed?

Comment (calebevans) [reply src/report/report.py]: @rnetser it's not needed. I originally was using it to filter any empty lines or lines with only whitespace. After some further testing, I found it works without. Removed.

**Review by myakove — APPROVED** [review]

**Pull request merged**
